app/src/main/java/study_dev/testbed/studyhopper/ui/sessions/CreateSession.java

```diff
@@ -0,0 +1,247 @@
+import android.app.TimePickerDialog;
+import android.content.Intent;
+import android.os.Bundle;
+import android.view.View;
+import android.widget.DatePicker;
+import android.widget.EditText;
+import android.widget.RadioButton;
+import android.widget.TimePicker;
+
+import androidx.appcompat.app.ActionBar;
+import androidx.appcompat.app.AppCompatActivity;
+
+import com.google.android.material.snackbar.Snackbar;
+import com.google.firebase.Timestamp;
+import com.google.firebase.firestore.CollectionReference;
+import com.google.firebase.firestore.DocumentReference;
+import com.google.firebase.firestore.FirebaseFirestore;
+
+import java.text.ParseException;
+import java.text.SimpleDateFormat;
+import java.util.Calendar;
+import java.util.Date;
+import java.util.Locale;
+
+import study_dev.testbed.studyhopper.R;
+import study_dev.testbed.studyhopper.models.Session;
+import study_dev.testbed.studyhopper.ui.studyGroup.StudyGroupActivity;
+
+public class CreateSession extends AppCompatActivity {
+//    private studyGroupItem item;
+    private EditText sessionNameEditText;
+    private EditText sessionDescriptionEditText;
+    private EditText sessionLocationEditText;
+    private EditText sessionDateEditText;
+    private EditText sessionTimeStartEditText, sessionTimeEndEditText;
+    private DatePickerDialog.OnDateSetListener dateListener;
+    private TimePickerDialog.OnTimeSetListener timeListener;
+    private DocumentReference groupDocRef;
+    private String userGroupId;
+    private String groupDocId;
+    private int sessionSelected = -1;
+
+    private FirebaseFirestore db = FirebaseFirestore.getInstance();
+    private CollectionReference sessionRef;
+
+    @Override
+    protected void onCreate(Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+        setContentView(R.layout.activity_create_session);
+
+        Intent intent = getIntent();
+        userGroupId = intent.getStringExtra("userGroupId");
+        groupDocId = intent.getStringExtra("groupId");
+
+        sessionRef = db.collection("groups").document(groupDocId).collection("sessions");
+
+
+        getSupportActionBar().setHomeAsUpIndicator(R.drawable.ic_close);
+        setTitle("Create a Study Session");
+        // Enable back button
+        ActionBar supportActionBar = getSupportActionBar();
+        if (supportActionBar != null) {
+            supportActionBar.setDisplayHomeAsUpEnabled(true);
+        }
+
+        sessionNameEditText = findViewById(R.id.edit_text_session_name);
+        sessionDescriptionEditText = findViewById(R.id.edit_text_session_description);
+        sessionLocationEditText = findViewById(R.id.edit_text_session_location);
+        sessionDateEditText = findViewById(R.id.edit_text_session_date);
+        sessionTimeStartEditText = findViewById(R.id.edit_text_session_start_time);
+        sessionTimeEndEditText = findViewById(R.id.edit_text_session_end_time);
+
+    }
+
+
+    public void selectSessionDate(View v) {
```

Comment:
Maybe would should abstract this into it's own class so we can use the same one for all calendars. The only thing is we would have to pass in the text boxes somehow.

Reply:
Yeah that's not a bad idea the dialogues are very useful for multiple areas of the app.